src/Network/NWProtocolTcpOptions.cs

```diff
@@ -0,0 +1,212 @@
+//
+// NWProtocolTcp: Bindings the Netowrk nw_protocol_options API focus on Tcp options.
+//
+// Authors:
+//   Manuel de la Pena <mandel@microsoft.com>
+//
+// Copyrigh 2019 Microsoft Inc
+//
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+using Security;
+using OS_nw_protocol_definition=System.IntPtr;
+using IntPtr=System.IntPtr;
+
+namespace Network {
+
+	[TV (12,0), Mac (10,14), iOS (12,0), Watch (6,0)]
+	public class NWProtocolTcpOptions : NWProtocolOptions {
+		// default values, same are used in swift
+		bool noDelay = false;
+		bool noPush = false;
+		bool noOptions = false;
+		bool enableKeepAlive = false;
+		uint keepAliveCount = 0;
+		TimeSpan keepAliveIdleTime = TimeSpan.Zero;
+		TimeSpan keepAliveInterval = TimeSpan.Zero;
+		uint maximumSegmentSize = 0;
+		TimeSpan connectionTimeout = TimeSpan.Zero;
+		TimeSpan persistTimeout = TimeSpan.Zero;
+		TimeSpan connectionDropTime = TimeSpan.Zero;
+		bool retransmitFinDrop = false;
+		bool disableAckStretching = false;
+		bool enableFastOpen = false;
+		bool disableEcn = false;
```

Comment:
you have the same issue here ^

and it seems https://github.com/xamarin/xamarin-macios/pull/7196/files#diff-2f2e76e5bddde26c68898ef0220c89adR84 needs to be updated to create the right (Tcp vs Udp) type